backend/inspirehep/records/api/mixins.py

```diff
@@ -637,3 +637,59 @@ def get_modified_experiment_uuids(self):
         pids_changed = set.symmetric_difference(set(pids_latest), set(pids_previous))
 
         return list(self.get_records_ids_by_pids(list(pids_changed)))
+
+
+class AddressMixin:
+    @classmethod
+    def create(cls, data, *args, **kwargs):
+        data = cls.build_address(data)
+        return super().create(data, *args, **kwargs)
+
+    def update(self, data, *args, **kwargs):
+        data = self.build_address(data)
+        super().update(data, *args, **kwargs)
+
+    @staticmethod
+    def get_geolocation(address):
+        response = geocoder.osm(address)
+        LOGGER.info(f"geocoder.osm response: {response}")
+        if response.ok:
+            latitude = response.current_result.lat
+            longitude = response.current_result.lng
+
+        return latitude, longitude
+
+    @staticmethod
+    def country_code_to_name(country_code):
+        try:
+            return country_code_to_name(country_code)
+        except KeyError:
+            return None
+
+    @classmethod
+    def build_address(cls, data):
+        for idx, address in enumerate(data.get("addresses", [])):
+            latitude = address.get("latitude")
+            longitude = address.get("longitude")
+            if latitude and longitude:
+                LOGGER.info("Using stored geolocation data.")
+                continue
+            postal_address = address.get("postal_address")[0]
+            if not postal_address:
+                place_name = address.get("place_name")
+                postal_code = address.get("postal_code")
+                city = address.get("cities", [None])[0]
+                state = address.get("state")
+                country = cls.country_code_to_name(address.get("country_code"))
+
+                address_list = [place_name, postal_code, city, state, country]
+                postal_address = ", ".join(
+                    [addr_part for addr_part in address_list if addr_part]
+                )
+            LOGGER.info(f"Querying OSM for address: {postal_address}")
+            latitude, longitude = cls.get_geolocation(postal_address)
+            if latitude and longitude:
+                LOGGER.info("Geolocation data has been found.")
+                data["addresses"][idx]["latitude"] = latitude
+                data["addresses"][idx]["longitude"] = longitude
```

Comment:
```suggestion
                address["latitude"] = latitude
                address["longitude"] = longitude
```
so you don't need to `enumerate`.

Reply:
:+1: